server/jetstream_api.go

```diff
@@ -836,6 +855,24 @@ func (s *Server) jsStreamCreateRequest(sub *subscription, c *client, subject, re
 	}
 
 	var resp = JSApiStreamCreateResponse{ApiResponse: ApiResponse{Type: JSApiStreamCreateResponseType}}
+
+	// Determine if we should proceed here when we are in clustered mode.
+	if s.JetStreamIsClustered() {
```

Comment:
This seem to be an exact copy of the code you added above, and looks like you do the same again in many different places. Should it be a function that maybe returns a boolean to indicate if calling code should proceed or not?

Reply:
I think so longer term. The response is different but I could see if I could reduce some of the repeated parts.. WDYT?

Reply:
I would make it a function because right now this is a lot of duplicate code. If long term one of the API need to be different then you don't call that function and have specific code. When the number of calls to this common function falls to 1 then we can remove :-)

Reply:
ok will do.

Reply:
Ok, I realize that the response objects are different.. sorry about that.

Reply:
But, they all embed ApiResponse and  jsonResponse() already works with an interface{}, so you could make that new function generic?

Reply:
Ok, we can always do that later. The code is there now, so that's ok.